tokens/src/lib.rs

```diff
@@ -444,7 +446,9 @@ impl<T: Config> Pallet<T> {
 				frame_system::Pallet::<T>::dec_consumers(who);
 			}
 		} else {
-			<Locks<T>>::insert(who, currency_id, locks);
+			let bounded_locks: BoundedVec<BalanceLock<T::Balance>, T::MaxLocks> =
+				locks.to_vec().try_into().expect("Too many locked balances");
```

Comment:
only use expect for things that can never fail. should modify the method signature to make it return `DispatchResult` instead